consensus/src/round_manager.rs

```diff
@@ -1296,6 +1353,75 @@ impl RoundManager {
         }
     }
 
+    async fn process_timeout_reception_result(
+        &mut self,
+        timeout: &RoundTimeout,
+        result: VoteReceptionResult,
+    ) -> anyhow::Result<()> {
+        let round = timeout.round();
+        match result {
+            VoteReceptionResult::NewQuorumCertificate(qc) => {
+                unreachable!()
+            },
+            VoteReceptionResult::New2ChainTimeoutCertificate(tc) => {
```

Comment:
we probably can split the vote from timeout in the pending_vote too, so the result can be separate (after we disable  the current vote msg)

Reply:
Will add a TODO to do this after disabling the current vote handling.